xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/macros.vm

```diff
@@ -1113,14 +1113,14 @@ $html
  *
  * @param docNames list of document names.
  * @param displaySpaces true to group documents by space.
- * @param blacklistedSpaces spaces to exclude from the list.
+ * @param ignoredSpaces spaces to exclude from the list.
  *#
-#macro(displayDocumentList $docNames $displaySpaces $blacklistedSpaces)
+#macro(displayDocumentList $docNames $displaySpaces $ignoredSpaces)
   #set($documentList = [])
   #foreach($docName in $docNames)
     #if($xwiki.hasAccessLevel("view", $xcontext.user, $docName))
       #set($document = $xwiki.getDocument($docName).getTranslatedDocument())
-      #if(("$!blacklistedSpaces" != '' && !$blacklistedSpaces.contains($document.getSpace())) || ("$!blacklistedSpaces" == '' && ($services.user.properties.displayHiddenDocuments() || !$document.isHidden())))
+      #if(("$!ignoredSpaces" != '' && !$ignoredSpaces.contains($document.getSpace())) || $services.user.properties.displayHiddenDocuments() || !$document.isHidden())
```

Comment:
Shouldn't the condition be: the space is not ignored **AND** either hidden documents are displayed or the document is not hidden?

```suggestion
      #if(!$ignoredSpaces.contains($document.getSpace()) && ($services.user.properties.displayHiddenDocuments() || !$document.isHidden()))
```

Which is close to what you initially wrote (was just missing the check for the hidden documents preference).